elide-model-config/src/test/java/com/yahoo/elide/modelconfig/validator/DynamicConfigValidatorTest.java

```diff
@@ -91,18 +91,60 @@ public void testMissingArgumentValue() throws Exception {
     public void testMissingConfigDir() throws Exception {
         String error = tapSystemErr(() -> {
             int exitStatus = catchSystemExit(() ->
-                    DynamicConfigValidator.main(new String[] { "--configDir", "src/test/resources/validator/missing" }));
+                    DynamicConfigValidator.main(new String[] { "--configDir", "src/test/resources/validator/missing", "--nocompile" }));
             assertEquals(2, exitStatus);
         });
 
         assertTrue(error.contains("config path does not exist"));
     }
 
+    @Test
+    public void testValidConfigDir() throws Exception {
+        tapSystemErr(() -> {
+            int exitStatus = catchSystemExit(() ->
+                    DynamicConfigValidator.main(new String[] { "--configDir", "src/test/resources/validator/valid", "--nocompile" }));
+            assertEquals(0, exitStatus);
+        });
+
+        tapSystemErr(() -> {
+            int exitStatus = catchSystemExit(() ->
+                    DynamicConfigValidator.main(new String[] { "--configDir", "src/test/resources/validator/valid", "--nomodelcompile", "--nodbcompile" }));
+            assertEquals(0, exitStatus);
+        });
+    }
+
+    @Test
+    public void testValidConfigDirWithCompile() throws Exception {
+        String error = tapSystemErr(() -> {
+            int exitStatus = catchSystemExit(() ->
+                    DynamicConfigValidator.main(new String[] { "--configDir", "src/test/resources/validator/valid" }));
+            assertEquals(2, exitStatus);
```

Comment:
This looks confusing.  Why do we expect this to fail since the directory is valid?

Reply:
hjson compilation will fail as it wont find classes defined in other modules.